internal/k8s/entity.go

```diff
@@ -78,45 +79,50 @@ func ImmutableEntities(entities []K8sEntity) []K8sEntity {
 	return result
 }
 
-type LoadBalancer struct {
+type LoadBalancerSpec struct {
 	Name  string
 	Ports []int32
 }
 
-func ToLoadBalancers(entities []K8sEntity) []LoadBalancer {
-	result := make([]LoadBalancer, 0)
+type LoadBalancer struct {
+	Spec LoadBalancerSpec
+	URL  *url.URL
+}
+
+func ToLoadBalancerSpecs(entities []K8sEntity) []LoadBalancerSpec {
+	result := make([]LoadBalancerSpec, 0)
 	for _, e := range entities {
-		lb, ok := ToLoadBalancer(e)
+		lb, ok := ToLoadBalancerSpec(e)
 		if ok {
 			result = append(result, lb)
 		}
 	}
 	return result
 }
 
-// Try to convert the current entity to a LoadBalancer service
-func ToLoadBalancer(entity K8sEntity) (LoadBalancer, bool) {
+// Try to convert the current entity to a LoadBalancerSpec service
+func ToLoadBalancerSpec(entity K8sEntity) (LoadBalancerSpec, bool) {
 	service, ok := entity.Obj.(*v1.Service)
```

Comment:
wait, do load balances only exist in v1?

Reply:
AFAIK, there's two ways to get traffic from outside the cluster to the pods
- v1.Service
- v1beta.Ingress

my vague understanding is that Ingress is still in flux and only really works on GKE, but i certainly haven't been following the discussions beyond the bare minimum